QA.md

```diff
@@ -0,0 +1,77 @@
+# Suggested Test Sequence
+
+1. **Simple Initialization**
+1. - [ ] Initialize a new wallet.
+1. - [ ] When creating a PIN, type the wrong PIN during the confirmation step.
+
+1. **Generate token error Tests**
+1. - [ ] Go to Settings and Create a new Token.
+1. - [ ] Enter token name Test Token, and click Next.
+1. - [ ] Enter token symbol TEST, and click Next.
+1. - [ ] Enter the amount of 100, and click Next.
+1. - [ ] Click on create token and check the error message.
+1. - [ ] Check that the token has not been created.
+
+1. **Receive Tests**
+1. - [ ] Go to the Receive Screen and check the QR Code.
+1. - [ ] Click on the address to copy it.
+1. - [ ] Click on New address and check that a new address was generated. The QRCode must update.
+1. - [ ] Click on Share and Cancel.
+1. - [ ] Click on Payment Request and generate a payment request of 10 HTR.
+1. - [ ] Pay the Payment Request using another wallet and check the confirmation message.
```

Comment:
The desktop wallet does not support reading QR codes nor payment requests. And this is really hard with a single phone. Should we improve our strategy?

Reply:
You can use the emulator on the desktop.

Reply:
The emulator can't read the generated payment request QRCode. The idea is to try to execute all the actions with the mobile phone but I guess in this case we can test this payment request with the emulator and then read with the phone.

Is it ok for now this @jansegre?

Reply:
Can't we leave an image with the QR Code that will be used when running the QA?

Reply:
I think the idea here is to validate two things: 

1. The QRcode is being correctly generated and if you read it, will be paid.
2. When the payment request is paid, it receives the payment from websocket and shows success on the screen.

If we have a fixed QRCode image neither of the points will be tested. I think for now the best option is to use the emulator.

Reply:
I think that requiring the QA tester to run an emulator is a big burden for setting up a testing environment.

Personally I have somewhat worked this around another way, my phone has a dual app feature that enables two independent instances of the same app, I use that in addition to taking a picture with my webcam of the qr code while I switch the app instances. It is a complicated setup that only works because of the dual app feature.

This step is testing two "features" at once:

- Being able to read and execute payment requests. A pre-generated image would cover this. But this would be a good additional step to the **Send Tests**.
- Validating the generated payment requests. The pre-generated image cannot test this.

I think it's good that we test executing the payment request with a different (but working) version of an existing wallet. This step doesn't specify this, but if we were to test using the same version (like I do on my setup, or maybe ends up happening on the emulator) we could run into a problem where we introduce a breaking change to the qr code data format which breaks older versions but the version being testes can read the qr code it generated. This would only be caught if the "other" wallet is an older version.

I don't have a good suggestion, but I have a workaround for this, so I guess it's fine.

Reply:
I think it is good for now. We can improve it later if necessary. Maybe we should consider that the QA must have access to two mobile devices to test them properly.